### PR TITLE
Project 2 submission

### PR DESCRIPTION
**Pull request by Osama-Tahboub1**
Includes Story 1 requirements and most of story 2's requirements.

### CODE REVIEW

File: about.php [hunk 1]
```diff
@@ -0,0 +1,99 @@
+<?php
+
+$db = new PDO('mysql:host=127.0.0.1; dbname=osamasCMSDB', 'root');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$aboutMeArticleQuery = $db->prepare("SELECT `aboutMeArticle` FROM `aboutMe`;");
```
Comment (charliecog) [inline about.php]: each query should be a function in a diff file that is called here

File: about.php [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+$aboutMeArticleQuery->execute();
+$aboutMeArticle = $aboutMeArticleQuery->fetchAll();
+
+$contactIconQuery = $db->prepare("SELECT `contactIcon` FROM `footer`");
```
Comment (charliecog) [inline about.php]: each query should be a function in a diff file that is called here

File: about.php [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+$contactIconQuery->execute();
+$contactIcon = $contactIconQuery->fetchAll();
+
+$smallContactIconQuery = $db->prepare("SELECT `smallContactIcon` FROM `footer`");
```
Comment (charliecog) [inline about.php]: each query should be a function in a diff file that is called here

File: about.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+$smallContactIconQuery->execute();
+$smallContactIcon = $smallContactIconQuery->fetchAll();
+
+$contactEmailQuery = $db->prepare("SELECT `contactEmail` FROM `footer`");
```
Comment (charliecog) [inline about.php]: each query should be a function in a diff file that is called here

File: about.php [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+$contactEmailQuery->execute();
+$contactEmail = $contactEmailQuery->fetchAll();
+
+$emailSubjectQuery = $db->prepare("SELECT `emailSubject` FROM `footer`");
```
Comment (charliecog) [inline about.php]: each query should be a function in a diff file that is called here

File: about.php [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+$emailSubjectQuery->execute();
+$emailSubject = $emailSubjectQuery->fetchAll();
+
+$adminLoginIconQuery = $db->prepare("SELECT `adminLoginIcon` FROM `footer`");
```
Comment (charliecog) [inline about.php]: each query should be a function in a diff file that is called here

File: about.php [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+$adminLoginIconQuery->execute();
+$adminLoginIcon = $adminLoginIconQuery->fetchAll();
+
+$adminLoginLinkQuery = $db->prepare("SELECT `adminLoginLink` FROM `footer`");
```
Comment (charliecog) [inline about.php]: each query should be a function in a diff file that is called here

File: about.php [hunk 8]
```diff
@@ -0,0 +1,99 @@
...
+$adminLoginLinkQuery->execute();
+$adminLoginLink = $adminLoginLinkQuery->fetchAll();
+
+/* Doc Block
```
Comment (charliecog) [inline about.php]: dont need to say Doc Block

File: about.php [hunk 9]
```diff
@@ -0,0 +1,99 @@
...
+ * Adds paragraph from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string the content of the arrays within the arrays pulled from the database.
```
Comment (charliecog) [inline about.php]: this function doesnt touch the array so not worth mentioning it in the docblock

File: cms/admin.php [hunk 10]
```diff
@@ -0,0 +1,25 @@
+<?php
+
+session_start();
+
+if ($_SESSION['loggedIn'] !== true) {
+    header('Location: instructions.php');
+}
+
+
```
Comment (charliecog) [inline cms/admin.php]: line break not needed

File: cms/admin.php [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <title>Content Management System</title>
+</head>
+<body>
+<nav>
+    <ul>
+        <li><a href="portfolio.php">Portfolio</a></li>
```
Comment (charliecog) [inline cms/admin.php]: what does this page do? if it is just a link to another page then delete this page and link there

File: cms/functions.php [hunk 12]
```diff
@@ -0,0 +1,260 @@
+<?php
+
+/* Doc Block
+ * Returns all project titles from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string a list of project titles in the arrays within the arrays pulled from the database.
```
Comment (charliecog) [inline cms/functions.php]: dont mention db

File: cms/functions.php [hunk 13]
```diff
@@ -0,0 +1,260 @@
...
+}
+
+/* Doc Block
+ * Returns all project title links from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string a list of project title links in the arrays within the arrays pulled from the database.
+ */
+function returnProjectTitleLinks(array $array):string {
```
Comment (charliecog) [inline cms/functions.php]: this is exactly the same as the function above - delete one of these and just have one function

File: about.php [hunk 14]
```diff
@@ -0,0 +1,99 @@
...
+ */
+function insertParagraph(array $array): string{
```
Comment (charliecog) [inline about.php]: same as function in admin.php

File: cms/functions.php [hunk 15]
```diff
@@ -0,0 +1,260 @@
...
+}
+
+/* Doc Block
+ * Returns project id's from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string a list of id's or keys in the arrays within the arrays pulled from the database.
+ */
+function returnProjectIds(array $array):string {
```
Comment (charliecog) [inline cms/functions.php]: duplication again. can be done programatically in function above

File: cms/functions.php [hunk 16]
```diff
@@ -0,0 +1,260 @@
...
+
+/* Doc Block
+ * Returns a single project title from array content provided by database.
+ *
+ * @param $array array associative array provided by database query.
+ * @param $recId int number pointing to the record id number in the database.
+ *
+ * @return string the title content for a single record.
+ */
+function returnProjectTitle(array $array, int $recId):string { // change title to record
```
Comment (charliecog) [inline cms/functions.php]: comment at the end of this line is messy - remove

Comment (charliecog) [inline cms/functions.php]: $array needs to be more verbose

File: cms/functions.php [hunk 17]
```diff
@@ -0,0 +1,260 @@
...
+    $index = $recId - 1;
```
Comment (charliecog) [inline cms/functions.php]: why?

File: cms/functions.php [hunk 18]
```diff
@@ -0,0 +1,260 @@
...
+    $projectTitleArray = $array[$index];
+    $projectTitle = '';
+    $projectTitle .=$projectTitleArray['projectTitle'];
```
Comment (charliecog) [inline cms/functions.php]: just return $projectTitleArray['projectTitle']; dont need the other lines

File: cms/functions.php [hunk 19]
```diff
@@ -0,0 +1,260 @@
...
+
+/* Doc Block
+ * Returns project title link from array content provided by database.
+ *
+ * @param $array array associative array provided by database query.
+ * @param $recId int number pointing to the record id number in the database.
+ *
+ * @return string the title link content for a single record.
+ */
+function returnProjectTitleLink(array $array, int $recId):string {
```
Comment (charliecog) [inline cms/functions.php]: same as function above

File: cms/functions.php [hunk 20]
```diff
@@ -0,0 +1,260 @@
...
+
+/* Doc Block
+ * Returns project image link from array content provided by database.
+ *
+ * @param $array array associative array provided by database query.
+ * @param $recId int number pointing to the record id number in the database.
+ *
+ * @return string the project image link for a single record.
+ */
+function returnProjectImage(array $array, int $recId):string {
```
Comment (charliecog) [inline cms/functions.php]: same as above

File: cms/functions.php [hunk 21]
```diff
@@ -0,0 +1,260 @@
...
+
+/* Doc Block
+ * Returns project id from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ * @param $recId int number pointing to the record id number in the database.
+ *
+ * @return string an id or key for a single record.
+ */
+function returnProjectId(array $array, int $recId):string {
```
Comment (charliecog) [inline cms/functions.php]: same as above

File: cms/functions.php [hunk 22]
```diff
@@ -0,0 +1,260 @@
...
+}
+
+/* Doc Block
+ * Returns article from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string the content of the arrays within the arrays pulled from the database.
+ */
+function returnArticle(array $array):string {
```
Comment (charliecog) [inline cms/functions.php]: same as the functions like returnProjectTitles etc...

File: cms/functions.php [hunk 23]
```diff
@@ -0,0 +1,260 @@
...
+}
+
+/* Doc Block
+ * Returns single paragraph from array content provided by database.
+ *
+ * @param $array array associative array provided by database.
+ *
+ * @return string the content of the arrays within the arrays pulled from the database.
+ */
+function returnParagraph(array $array, int $id): string{
```
Comment (charliecog) [inline cms/functions.php]: same as above

File: cms/functions.php [hunk 24]
```diff
@@ -0,0 +1,260 @@
...
+ * @param $projectImage string object used with post from form and passed in to update the portfolio table in database.
+ * @param $projectId string used with post from form and passed in to update the portfolio table in database.
+ *
+ * @return array returns associative array with arrays containing content from portfolio table from database.
+ */
+function updateProject(PDO $db, string $projectTitle, string $projectTitleLink, string $projectImage, string $projectId) {
+
+    $query = $db->prepare("UPDATE `portfolio` SET `projectTitle` = :projectTitle, `projectImage` = :projectImage, `projectTitleLink` = :projectTitleLink WHERE `id` = :projectId");
+
+    $query->bindParam (':projectTitle',$projectTitle);
```
Comment (charliecog) [inline cms/functions.php]: space between , and the next param

File: cms/functions.php [hunk 25]
```diff
@@ -0,0 +1,260 @@
...
+ *
+ * @param $db object database link and credentials PDO object.
+ * @param $projectTitle string used with post from form and passed in to update the portfolio table in database.
+ * @param $projectTitleLink string used with post from form and passed in to update the portfolio table in database.
+ * @param $projectImage string object used with post from form and passed in to update the portfolio table in database.
+ * @param $projectId string used with post from form and passed in to update the portfolio table in database.
+ *
+ * @return array returns associative array with arrays containing content from portfolio table from database.
+ */
+function updateProject(PDO $db, string $projectTitle, string $projectTitleLink, string $projectImage, string $projectId) {
```
Comment (charliecog) [inline cms/functions.php]: project ID as a string?

File: cms/functions.php [hunk 26]
```diff
@@ -0,0 +1,260 @@
...
+    $query = $db->prepare("UPDATE `portfolio` SET `projectTitle` = :projectTitle, `projectImage` = :projectImage, `projectTitleLink` = :projectTitleLink WHERE `id` = :projectId");
+
+    $query->bindParam (':projectTitle',$projectTitle);
+    $query->bindParam (':projectTitleLink',$projectTitleLink);
+    $query->bindParam (':projectImage',$projectImage);
+    $query->bindParam (':projectId',$projectId);
+
+    $query->execute();
+
+    $updatedInfo = $projectId.' '. $projectTitle.' '.$projectTitleLink.' '.$projectImage;
```
Comment (charliecog) [inline cms/functions.php]: this is a false positive - if your query fails then you would return data that hasnt actually gone in the db

File: cms/functions.php [hunk 27]
```diff
@@ -0,0 +1,260 @@
...
+    return $updatedInfo;
+}
+
+/* Doc Block
+ * Gets user credentials from the database.
+ *
+ * @param $db object database link and credentials PDO object.
+
```
Comment (charliecog) [inline cms/functions.php]: line break doesnt need to be here

File: cms/functions.php [hunk 28]
```diff
@@ -0,0 +1,260 @@
...
+ *
+ * @return array returns associative array with arrays credentials from users and passwords tables from database.
+ */
+function getUserCredentials(PDO $db):array {
+    $userCredentialsQuery = $db->prepare("SELECT `users`.`name`, `password` FROM `users` LEFT JOIN `passwords` ON `users`.`id` = `passwords`.`userId`");
+    $userCredentialsQuery->execute();
+    return $userCredentials = $userCredentialsQuery->fetchAll();
```
Comment (charliecog) [inline cms/functions.php]: just return $userCredentialsQuery->fetchAll();

File: cms/functions.php [hunk 29]
```diff
@@ -0,0 +1,260 @@
...
+ * @param $actualUserName string username stored credential.
+ * @param $actualPassword string password stored credential.
+ * @param $inputUserName string form inout username.
+ * @param $inputPassword string form input password.
+ *
+ * @return bool returns true if both content types are string.
+ */
+function checkCredentials(string $actualUserName, string $actualPassword, string $inputUserName, string $inputPassword): bool {
+
+
```
Comment (charliecog) [inline cms/functions.php]: line break doesnt need to be here

File: cms/functions.php [hunk 30]
```diff
@@ -0,0 +1,260 @@
...
+        return true;
+    } else {
+        return false;
+    }
+}
+
+/* Doc Block
+ * Directs to admin page or echoes instructions depending on credentials check.
+ *
+ *  @param $actualUserName string username stored credential.
```
Comment (charliecog) [inline cms/functions.php]: indentation

File: cms/functions.php [hunk 31]
```diff
@@ -0,0 +1,260 @@
...
+ * @param $actualPassword string password stored credential.
+ * @param $inputUserName string form inout username.
```
Comment (charliecog) [inline cms/functions.php]: spelling: input

File: cms/functions.php [hunk 32]
```diff
@@ -0,0 +1,260 @@
...
+ * @param $inputPassword string form input password.
+ *
+ * @return bool returns true if both content types are string.
+ */
+
+
+function loggedInSession(string $actualUserName, string $actualPassword, string $inputUserName, string $inputPassword ):bool {
+    $loginCheck = checkCredentials($actualUserName, $actualPassword, $inputUserName, $inputPassword);
```
Comment (charliecog) [inline cms/functions.php]: best practice not to call functions inside other functions. abstract out and pass in result of function call into this function

File: cms/functions.php [hunk 33]
```diff
@@ -0,0 +1,260 @@
...
+
+    if ($loginCheck === true) {
+        session_start();
```
Comment (charliecog) [inline cms/functions.php]: you could just put this functionality inside checkCredentials

File: cms/instructions.php [hunk 34]
```diff
@@ -0,0 +1,2 @@
+<?php
+echo 'Please log in';
```
Comment (charliecog) [inline cms/instructions.php]: put inside a function

Comment (charliecog) [reply cms/instructions.php]: also missing closing php tag

File: cms/login.php [hunk 35]
```diff
@@ -0,0 +1,38 @@
...
+$actualPassword = $credentials[0]['password'];
+
+$inputUserName = $_POST['userName'];
+$inputPassword = $_POST['password'];
+
+if ($inputUserName != NULL && $inputPassword != NULL) {
+    loggedInSession($actualUserName, $actualPassword, $inputUserName, $inputPassword);
+}
+
+
```
Comment (charliecog) [inline cms/login.php]: line break

File: cms/portfolio.php [hunk 36]
```diff
@@ -0,0 +1,238 @@
...
+require_once 'functions.php';
+
+$projectInfo = getProjectInfo($db);
+
+$projectTitle1 = $_POST['projectTitle1'];
+$projectTitleLink1 = $_POST['projectTitleLink1'];
+$projectImage1 = $_POST['projectImage1'];
+$projectId1 = $_POST['projectId1'];
+
+if ($projectTitle1 != NULL && $projectTitleLink1 != NULL && $projectImage1 != NULL && $projectId1!= NULL) {
```
Comment (charliecog) [inline cms/portfolio.php]: make into function

File: cms/portfolio.php [hunk 37]
```diff
@@ -0,0 +1,238 @@
...
+
+}
+
+
+$projectTitle2 = $_POST['projectTitle2'];
+$projectTitleLink2 = $_POST['projectTitleLink2'];
+$projectImage2 = $_POST['projectImage2'];
+$projectId2 = $_POST['projectId2'];
+
+if ($projectTitle2 != NULL && $projectTitleLink2 != NULL && $projectImage2 != NULL && $projectId2!= NULL) {
```
Comment (charliecog) [inline cms/portfolio.php]: use a function

File: cms/portfolio.php (hunk 36 above)
Comment (charliecog) [reply cms/portfolio.php]: same for all of the repeated code in this file

File: cms/portfolio.php [hunk 38]
```diff
@@ -0,0 +1,238 @@
...
+<head>
+    <meta charset="UTF-8">
+    <title>Portfolio</title>
+</head>
+<body>
+
+<h1>Portfolio</h1>
+
+    <div>
+        <h2>Current Values</h2>
```
Comment (charliecog) [inline cms/portfolio.php]: use a function that creates these for you and does it with a loop to save repeated code

Comment (charliecog) [reply cms/portfolio.php]: same for all in this file

File: cms/tests/functions.php [hunk 39]
```diff
@@ -0,0 +1,88 @@
...
...0=>["id"=>"1","projectTitle"=>"Pilot Shop","projectTitleLink"=>"https://dev.maydenacademy.co.uk/students/2018/osama/PilotShopSite/","projectImage"=>"https://dev.maydenacademy.co.uk/students/2018/osama/Portfolio/assets/PilotShop.png"],1=>["id"=>"2","projectTitle"=>"CMS Project","projectTitleLink"=>"https://dev.maydenacademy.co.uk/students/2018/osama/PilotShopSite/", "projectImage"=>"https://dev.maydenacademy.co.uk/students/2018/osama/Portfolio/assets/PilotShop.png"],2=>["id"=>"3","projectTitle"=>"New Project","projectTitleLink"=> "https://dev.maydenacademy.co.uk/students/2018/osama/PilotShopSite/","projectImage"=>"https://dev.maydenacademy.co.uk/students/2018/osama/Portfolio/assets/PilotShop.png"],3=>["id"=>"4","projectTitle"=>NULL,"projectTitleLink"=>NULL,"projectImage"=>NULL],4=>["id"=>"5","projectTitle"=> NULL,"projectTitleLink"=>NULL,"projectImage"=>NULL],5=>["id"=>"6","projectTitle"=>NULL,"projectTitleLink"=>NULL,"projectImage"=>NULL],6=>["id"=>"7","projectTitle"=>NULL,"projectTitleLink"=>NULL,"projectImage"=>NULL],7=>["id"=>"8","projectTitle"=>NULL,"projectTitleLink"=>NULL,"projectImage"=>NULL],8=>["id"=>"9","projectTitle"=>NULL,"projectTitleLink"=>NULL,"projectImage"=>NULL]];
```
Comment (charliecog) [inline cms/tests/functions.php]: can make the input array shorter as it only has to be dummy data

Comment (charliecog) [reply cms/tests/functions.php]: same for all tests

Comment (charliecog) [reply cms/tests/functions.php]: these tests will all become one when you change your functions into one in your code

File: index.php [hunk 40]
```diff
@@ -0,0 +1,70 @@
+<?php
+
+$db = new PDO('mysql:host=127.0.0.1; dbname=osamasCMSDB', 'root');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$contactIconQuery = $db->prepare("SELECT `contactIcon` FROM `footer`");
```
Comment (charliecog) [inline index.php]: make into functions

Comment (charliecog) [reply index.php]: all of these in this file

Comment (charliecog) [reply index.php]: actually just select out all the data you need at once

Comment (charliecog) [reply index.php]: eg: SELECT thing1, thing2, thing3 etc...

File: index.php [hunk 41]
```diff
@@ -0,0 +1,70 @@
...
+            <li><a href="portfolio.php">Portfolio</a></li>
+        </ul>
+    </nav>
+
+    <h1 class="homePageTitle">Osama's Portfolio</h1>
+</header>
+
+<footer class="container">
+    <div class="contactIcon">
+        <span>Contact me</span>
```
Comment (charliecog) [inline index.php]: use a function that loops over results and generates your two contact sections

File: portfolio.php [hunk 42]
```diff
@@ -0,0 +1,129 @@
+<?php
+
+$db = new PDO('mysql:host=127.0.0.1; dbname=osamasCMSDB', 'root');
+$db->setAttribute(PDO::ATTR_DEFAULT_FETCH_MODE, PDO::FETCH_ASSOC);
+
+$projectTitleQuery = $db->prepare("SELECT `projectTitle` FROM `portfolio`");
```
Comment (charliecog) [inline portfolio.php]: same as index.php file - do all selects in one query

File: about.php (hunk 9 above)
Comment (Osama-Tahboub1) [reply about.php]: changed to  * @return string the text content of an array pulled from the database.

File: about.php (hunk 8 above)
Comment (Osama-Tahboub1) [reply about.php]: removed.

File: about.php (hunk 1 above)
Comment (Osama-Tahboub1) [reply about.php]: done.

File: about.php (hunk 2 above)
Comment (Osama-Tahboub1) [reply about.php]: done.

File: about.php (hunk 3 above)
Comment (Osama-Tahboub1) [reply about.php]: done.

File: about.php (hunk 4 above)
Comment (Osama-Tahboub1) [reply about.php]: done.

File: about.php (hunk 5 above)
Comment (Osama-Tahboub1) [reply about.php]: done.

File: about.php (hunk 6 above)
Comment (Osama-Tahboub1) [reply about.php]: done.

File: about.php (hunk 7 above)
Comment (Osama-Tahboub1) [reply about.php]: done.

File: about.php (hunk 14 above)
Comment (Osama-Tahboub1) [reply about.php]: fixed by adding all function to one file then requiring it.

File: cms/admin.php (hunk 10 above)
Comment (Osama-Tahboub1) [reply cms/admin.php]: Line break removed.

File: cms/admin.php (hunk 11 above)
Comment (Osama-Tahboub1) [reply cms/admin.php]: This page used to host navigation links for multiple functionalities,  most of it removed due to MVP, will still built up on it as a navigator for further CMS functionality.

File: cms/functions.php (hunk 13 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Added one function for whole form with all default values.

File: cms/functions.php (hunk 12 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: db removed from documentation.

File: cms/functions.php (hunk 15 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Added one function for whole form with all default values.

File: cms/functions.php (hunk 16 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: removed.

Comment (Osama-Tahboub1) [reply cms/functions.php]: updated to projectInfoArray.

File: cms/functions.php (hunk 17 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Zero index array relation ship with record Id's from database.

File: cms/functions.php (hunk 18 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Edited for all 4 functions that use this.

File: cms/functions.php (hunk 19 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Edited for all 4 functions that use this.

File: cms/functions.php (hunk 20 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Edited for all 4 functions that use this.

File: cms/functions.php (hunk 21 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Edited for all 4 functions that use this.

File: cms/functions.php (hunk 23 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: summarised to 2 lines.

File: cms/functions.php (hunk 24 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Amended all 4 lines.

File: cms/functions.php (hunk 26 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Changed to return result of function that gets data from the database after the query is executed.

File: cms/functions.php (hunk 27 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Line break removed.

File: cms/functions.php (hunk 28 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Done.

File: cms/functions.php (hunk 29 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: This is removed.

File: cms/functions.php (hunk 30 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: Removed extra space.

File: cms/functions.php (hunk 31 above)
Comment (Osama-Tahboub1) [reply cms/functions.php]: This is fixed.

File: index.php (hunk 41 above)
Comment (Osama-Tahboub1) [reply index.php]: Done.

File: portfolio.php (hunk 42 above)
Comment (Osama-Tahboub1) [reply portfolio.php]: Done.

**Pull request merged**